src/compiler/moduleSpecifiers.ts

```diff
@@ -3,11 +3,12 @@
 namespace ts.moduleSpecifiers {
     export interface ModuleSpecifierPreferences {
         importModuleSpecifierPreference?: "relative" | "non-relative";
+        includeExtensionInImports?: Extension.Js | Extension.Jsx;
     }
 
     // Note: fromSourceFile is just for usesJsExtensionOnImports

```

Comment:
Copied from https://github.com/Microsoft/TypeScript/pull/25074#discussion_r197240892 -- maybe we shouldn't take `fromSourceFile` as input and just use `includeExtensionInImports` for that.